crates/volta-migrate/src/v2.rs

```diff
@@ -0,0 +1,158 @@
+use std::convert::TryFrom;
+use std::fs::{read_to_string, remove_file, rename, write, File};
+use std::path::PathBuf;
+
+use super::empty::Empty;
+use super::v1::V1;
+use log::debug;
+use tempfile::tempdir_in;
+use volta_core::error::ErrorDetails;
+use volta_core::fs::{ensure_dir_does_not_exist, read_dir_eager};
+use volta_core::tool::load_default_npm_version;
+use volta_core::toolchain::serial::Platform;
+use volta_core::version::parse_version;
+use volta_fail::{Fallible, ResultExt, VoltaError};
+use volta_layout::v2;
+
+/// Represents a V2 Volta Layout (from v0.7.3)
```

Comment:
Two notes:

1. Can you clarify (both in the PR and in this comment) what "from v0.7.3" means?
2. If it means "available from 0.7.3 forward", should it actually be v0.8.0? Technically the layout internally is private API, so I can see an argument either way, but it's a change that seems breaking-ish, in that anyone managing things manually has to run `volta setup` again to migrate?

Reply:
Yes, it's intended to mean "available from 0.7.3 forward", I'll update the comment to be more clear about that.

To the other question, the migrations are checked on any run of `volta` or a shim, so there isn't a need to run specifically `volta setup` again. Once the user has the new binaries, the migration will run the next time they interact with Volta.

We also don't support "managing things manually" when it comes to the data directory, things have to be laid out exactly how Volta expects them or it won't be able to find what it's looking for. The migrations allow us to make those changes without requiring a "breaking" change the way we used to (where we required the user to completely delete their installation and re-install), instead we make any updates in-place.

Reply:
Thanks for clarifying!